File: Pattern/chopper2DArray.c
#include <stdio.h>

int main() {
    int grid[8][8] = {0};  // Create an 8x8 grid to represent the houses

    // Input the number of choppers
    int numChoppers;
    printf("How many choppers to land? ");
    scanf("%d", &numChoppers);

    // Input the coordinates of each helicopter
    for (int i = 0; i < numChoppers; i++) {
        int x, y;
        printf("Enter the coordinates to land the chopper [x,y = 1-8](x y): " );
        scanf("%d %d", &x, &y);
        grid[x-1][y-1] = 1;  // Mark the house with a terrorist
    }

    int terroristsKilled = 0;

    // Count the number of terrorists killed
    for (int i = 0; i < 8; i++) {
        for (int j = 0; j < 8; j++) {
            if (grid[i][j] == 1) {
                terroristsKilled++;
            }
        }
    }

    printf("%d terrorists have been killed in total.\n", terroristsKilled);

    return 0;
}

File: Pattern/chopper2DArray.py
def compute_terrorists_killed(num_choppers):
    grid = [[0] * 8 for _ in range(8)]  # Create an 8x8 grid to represent the houses
    terrorists_killed = 0

    # Input the coordinates of each helicopter
    for i in range(num_choppers):
        x, y = map(int, input("Enter the coordinates to land the chopper [x,y = 1-8] #" + str(i+1) + " (x,y): ").split(","))
        grid[x-1][y-1] = 1  # Mark the position of the helicopter on the grid

    # Count the number of terrorists killed
    for i in range(8):
        row_killed = 0
        col_killed = 0
        for j in range(8):
            if grid[i][j] == 1:
                row_killed = 1
            if grid[j][i] == 1:
                col_killed = 1
        terrorists_killed += row_killed + col_killed

    return terrorists_killed

num_choppers = int(input("How many choppers to land? "))
terrorists_killed = compute_terrorists_killed(num_choppers)
print(terrorists_killed, "terrorists have been killed in total.")
